src/wallet/test/util.h

```diff
@@ -116,7 +116,7 @@ class MockableDatabase : public WalletDatabase
 std::unique_ptr<WalletDatabase> CreateMockableWalletDatabase(MockableData records = {});
 MockableDatabase& GetMockableDatabase(CWallet& wallet);
 
-ScriptPubKeyMan* CreateDescriptor(CWallet& keystore, const std::string& desc_str, const bool success);
+std::optional<std::reference_wrapper<DescriptorScriptPubKeyMan>> CreateDescriptor(CWallet& keystore, const std::string& desc_str, const bool success);
```

Comment:
In commit "wallet: Use util::Error throughout AddWalletDescriptor" (218db39973736ac50e912bec440b1e512586d3b0)

I think I understand the need to use std::reference_wrapper with util::Result. But it would seem good to replace `std::optional<std::reference_wrapper<DescriptorScriptPubKeyMan>>` here and other places with just `DescriptorScriptPubKeyMan*` for simplicity and avoid reference_wrapper except in the cases where it's really needed. It seems like doing this would make a lot of the test changes here unnecessary and the tests themselves more readable. 

I think use of reference_wrapper could to go away later with #25665 and normal references could be used.

Reply:
Done as suggested

Reply:
re: https://github.com/bitcoin/bitcoin/pull/32475#discussion_r2098230420

> I guess `reference_wrapper` then is still needed in CWallet::AddWalletDescriptor?

Yes it's still probably better to use reference_wrapper there instead of using a pointer type which could be null. I think more ideally util::Result would support reference types, but it doesn't currently. (When I wrote previous comment I though #25665 would allow util::Result to support reference types, but that's not actually true. It could help, but would not be sufficient on its own because the implementation uses a union and references aren't allowed in unions.)